internal/integration/api/serviceaccount.go

```diff
@@ -0,0 +1,293 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//go:build integration_api
+// +build integration_api
+
+package api
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/siderolabs/go-pointer"
+	"github.com/talos-systems/go-retry/retry"
+	corev1 "k8s.io/api/core/v1"
+	eventsv1 "k8s.io/api/events/v1"
+	kubeerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/talos-systems/talos/internal/integration/base"
+	machineapi "github.com/talos-systems/talos/pkg/machinery/api/machine"
+	"github.com/talos-systems/talos/pkg/machinery/client"
+	"github.com/talos-systems/talos/pkg/machinery/client/config"
+	"github.com/talos-systems/talos/pkg/machinery/config/types/v1alpha1"
+	"github.com/talos-systems/talos/pkg/machinery/config/types/v1alpha1/machine"
+)
+
+var (
+	serviceAccountGVR = schema.GroupVersionResource{
+		Group:    "talos.dev",
+		Version:  "v1alpha1",
+		Resource: "serviceaccounts",
```

Comment:
should we use constants here?

Reply:
I prefer to not use variables in tests so that they would fail explicitly if we change a constant and I think that's ok&safer. String literals > vars in tests. Wdyt?

Reply:
I feel it might be some sort of confusion eventually. This way tests would fail if we change the constants, but probably tracking it down won't be trivial, and it might seem that we don't have constants for those value.

I would suggest to keep using constants, as if we change them, we know we changed them :)

Reply:
done